### ISSUE TITLE
Is it false-positive?

### ISSUE THREAD

**Issue by b1rdex** [kalessil/phpinspectionsea#338]
```php
/**
 * @property \Sp\Cache\RedisCache                                $cache
 */
class WebApplication extends \CWebApplication
{}

class RedisCache extends PredisCacheAdapter
{}

abstract class PredisCacheAdapter extends CCache
{}

abstract class CCache extends CApplicationComponent implements ICache, ArrayAccess
{}

class ActionLock
{
    public function __construct(int $userId, ICache $cache = null)
    {
        if ($cache === null) {
            $cache = Yii::app()->cache; // this is marked — \Sp\Cache\RedisCache is not in annotated types
        }
    }
```

Sorry I can't extract minimal reproducible example. I tried to replay this defining function and class and interface, but problem gone. May be problem appears only with `@property` annotation or classes.

**Comment by kalessil** [comment]
That's a false-positive (or some work ahead as I can judge xD) - the inspection probably don't check the classes hierarchy.

**Comment by kalessil** [comment]
Sooooo, I had to refactor the inspection and add missing UTs - this done.
Now the question if this issue has gone away or not: @b1rdex can you give a try to a fresh binary from our master?

PS: I'll re-open ticket in case of negative feedback.

**Comment by b1rdex** [comment]
Issue gone.

**Comment by kalessil** [comment]
Awesome, thank you for checking.
